src/components/WatchVideo.vue

```diff
@@ -540,16 +540,21 @@ export default {
         },
         handleClick(event) {
             if (!event || !event.target) return;
-            var target = event.target;
-            if (
-                !target.nodeName == "A" ||
-                !target.getAttribute("href") ||
-                !target.innerText.match(/(?:[\d]{1,2}:)?(?:[\d]{1,2}):(?:[\d]{1,2})/)
-            )
-                return;
-            const time = parseInt(target.getAttribute("href").match(/(?<=t=)\d+/)[0]);
+            if (!event.target.matches("a[href]")) return;
+            const target = event.target;
+            if (!target.getAttribute("href")) return;
+            if (this.handleTimestampLinks(target)) {
+                event.preventDefault();
+            }
+        },
+        handleTimestampLinks(target) {
+            const url = new URL(target.getAttribute("href"), document.baseURI);
+            if (!url.searchParams.has("t")) return false;
+            const time = parseInt(url.searchParams.get("t"));
```

Comment:
It may not be safe to assume `t` is a number. I remember there was a function somewhere to parse this.